src/overlays/actors/ovl_Obj_Timeblock/z_obj_timeblock.c

```diff
@@ -27,41 +57,314 @@ const ActorInit Obj_Timeblock_InitVars = {
+        // "◯◯◯◯ Time Block Attention Camera (frame counter  %d)\n"
+        osSyncPrintf("◯◯◯◯ Time Block 注目カメラ (frame counter  %d)\n", globalCtx->state.frames);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA032C.s")
+        ObjTimeblock_ToggleSwitchFlag(globalCtx, this->dyna.actor.params & 0x3F);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA040C.s")
+    func_80BA06AC(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA0480.s")
+    if (this->demoEffectTimer == 50) {
+        func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+    }
+    if (!this->isVisible && this->demoEffectTimer <= 0) {
+        ObjTimeblock_StartAltBehaviourNotVisibleActionFunc(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA04F8.s")
+void ObjTimeblock_StartAltBehaviourNotVisibleActionFunc(ObjTimeblock* this) {
+    this->actionFunc = &ObjTimeblock_AltBehaviourNotVisibleActionFunc;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA0508.s")
+void ObjTimeblock_AltBehaviourNotVisibleActionFunc(ObjTimeblock* this, GlobalContext* globalCtx) {
+    s32 switchFlag = this->dyna.actor.params & 0x3F;
+    s8 switchFlagIsSet = BOOL(Flags_GetSwitch(globalCtx, switchFlag));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA0514.s")
+    if (this->unk_176 ^ switchFlagIsSet && switchFlagIsSet ^ BOOL((this->dyna.actor.params >> 15) & 1)) {
+        if (this->demoEffectTimer <= 0) {
+            ObjTimeblock_SpawnDemoEffect(this, globalCtx);
+            this->demoEffectTimer = 160;
+        }
+        this->demoEffectFirstPartTimer = 12;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA0524.s")
+    func_80BA06AC(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA06AC.s")
+    if (this->isVisible && this->demoEffectTimer <= 0) {
+        ObjTimeblock_StartAltBehaviorVisibleActionFunc(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA0758.s")
+void ObjTimeblock_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjTimeblock* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA0768.s")
+    this->actionFunc(this, globalCtx);
+    if (this->demoEffectTimer > 0) {
+        this->demoEffectTimer--;
+    }
+    if (this->isVisible) {
+        func_8003EC50(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    } else {
+        func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA083C.s")
+void ObjTimeblock_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    if (THIS->isVisible) {
```

Comment:
Can you afford a `this = THIS` temp?

Reply:
no stack space oddly enough, couldnt get rid of the color temp either